File: solutions/TypeScript/1.TwoSum/main.ts
import { twoSum } from "./TwoSum";
import { nums, target } from "./GenerateRandomTargetAndArray";

const result = twoSum(nums, target);

function printResult(result: number[]): void {
  if (result.length === 0) {
    console.log(`No pair of numbers adds up to the target sum of ${target}.`);
  } else {
    console.log(
      `The addition of nums[${result[0]}]:${nums[result[0]]} and nums[${
        result[1]
      }]:${nums[result[1]]} is ${target}`
    );
  }
}

printResult(result);

File: solutions/Python/1.TwoSum/main.py
from TwoSum import (
    Solution,
)
from GenerateRandomTargetAndArray import nums, target


def print_result(result, target, nums):
    if len(result) == 0:
        print(f"No pair of numbers adds up to the target sum of {target}.")
    else:
        print(
            f"The addition of nums[{result[0]}]:{nums[result[0]]} and nums[{result[1]}]:{nums[result[1]]} is {target}"
        )


solution = Solution()
result = solution.twoSum(nums, target)
print_result(result, target, nums)
